coursera_login.py

```diff
@@ -0,0 +1,100 @@
+'''
+Created on Nov 5, 2012
+
+@author: alex
+'''
+import random
+import string
+
+from xbmcswift2 import Plugin
+import requests
+
+plugin = Plugin()
+
+BASE_CLASS_URL = "https://class.coursera.org"
+LOGIN_URL = "https://accounts.coursera.org/api/v1/login"
+
+
+def makeCSRFToken():
+    csrftoken = ''.join(random.choice(string.ascii_uppercase + string.digits +
+                                      string.ascii_lowercase)
+                        for _ in xrange(24))
+    return csrftoken
+
+
+def login(username, password):
+    csrftoken = makeCSRFToken()
+    headers = {
+        'Referer': 'https://www.coursera.org/account/signin',
+        'Host': "accounts.coursera.org",
+        'X-CSRFToken': csrftoken,
+        'X-Requested-With': 'XMLHttpRequest',
+        'Origin': 'https://accounts.coursera.org',
+        'Cookie': 'csrftoken=%s' % csrftoken,
+    }
+    data = dict(email=username, password=password, webrequest='true')
+    res = requests.post(LOGIN_URL, data=data, headers=headers)
+    assert res.ok, "Could not login: %s" % res.content
+    return res.cookies.get_dict()
+
+
+def get_auth_url(className):
+    return ("%s/%s/auth/auth_redirector?type=login&"
+            "subtype=normal&email=&visiting=&minimal=true") % (
+                BASE_CLASS_URL, className)
+
+
+@plugin.cached()
+def getClassCookies(className, username, password):
+    user_data = plugin.get_storage(username, file_format='json')
+    if user_data is None:
+        return None
+
+    cookies_raw = user_data.get('cookies')
+    if cookies_raw is None:
+        cookies_raw = login(username, password)
+        if cookies_raw is None:
+            return None
+        user_data['cookies'] = cookies_raw
+
+    res = requests.get(get_auth_url(className), allow_redirects=False,
+                       cookies=cookies_raw)
+    if not res.ok:
```

Comment:
The `res.ok` check is redundant (see: https://github.com/kennethreitz/requests/blob/bd3cf95e34aa49c8d764c899672048df107e0d70/requests/models.py#L617). Just call `res.raise_for_status()`.